lib/ansible/plugins/terminal/ios.py

```diff
@@ -58,7 +58,11 @@ def on_open_shell(self):
             raise AnsibleConnectionFailure('unable to set terminal parameters')
 
     def on_become(self, passwd=None):
-        if self._get_prompt().endswith(b'#'):
+        conn_prompt = self._get_prompt()
+        if conn_prompt:
```

Comment:
After thinking this through, this whole nested construct seems to be replaceable with:
```python
if conn_prompt or conn_prompt.endswith(b'#'):
    return
```
right?

Reply:
yea, that's right thanks and have I updated the PR